api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -301,6 +302,10 @@ private User initializeUserIfNeeded() {
         return userDao.save(user);
 
       case READY:
+        try {
+          this.notebooksService.createCluster(
+              user.getFreeTierBillingProjectName(), NotebooksServiceImpl.getDefaultClusterName(), user.getEmail());
+        } catch (ConflictException e) {}
```

Comment:
Per the associated ticket, my primary concern with doing this is that it would time out and/or make the initial /getMe very slow. Are you not seeing that happen?

Reply:
It definitely does time out... however, we don't get slowness on the initial /getMe, because it is only called when the billing project is ready, and only that once. Thus, it happens in the polling cycle, rather than the initial load. If a user: created their account, logged in, then closed their window before the billing project was ready, and _then_ signed back in, it would be slow.

Reply:
We probably shouldn't bake in an assumption around whether `/getMe` is going to block the UI or not, though agreed that currently it's likely to be during the polling loop. Do you see a 500 in the left-hand corner when this happens?

Reply:
Regardless of the above, I think we should probably explore the outgoing request timeout or sending a non-blocking request for cluster create and do it if it's easy.

Reply:
We do see the 500 in the left hand corner when it happens.

Reply:
I'd be in favor of making the call to createCluster() asynchronous... unless we need to get some information back from it?

Reply:
Is there a way to make an asynchronous call from Java like that?

Reply:
You could try the `createClusterAsync()` method ;)

But in seriousness, I'm not sure whether that will work on app engine. Alternatively, you may have more control if you build the call directly, see the Swagger client.

We should also figure out whether we're OK just trying this once and assuming it worked, or if we want to be retrying until we're confident the cluster got created.

Reply:
Yeah... I'm a little bit wanting to block on anything that's not a conflict or a socket timeout. And if socket timeout and then it doesn't work then we try again on hitting workspace.

Reply:
So you can't choose to block on some things but not others.

You could choose to call "apiClient.setConnectionTimeout(1000); // millis"

and then you'll get an exception faster than the default (which appears to be ten seconds) if Leo doesn't respond.

I just figured that we should send off the request asynchronously, and when checking for cluster status re-send if we discover that Leo doesn't know anything about our cluster.

I believe that in the latest version of AppEngine (that supports Java 8), async stuff should generally work fine, see:

https://cloud.google.com/appengine/docs/standard/java/runtime-java8

Reply:
Note: I don't know when you call an async method whether Swagger tries to connect before returning, or if it's entirely asynchronous. We should investigate.

Reply:
I did some digging, sounds like it doesn't attempt to connect, but instead schedules a request to be sent asynchronously. So if we use the async method we won't have any notion that something is wrong with the request or their server in the response. But arguably the client doesn't have to care about that... as long as we log when it happens?

Reply:
That may honestly be fine. I'll poke around when I get in on Monday?

Reply:
SGTM!

Reply:
Follow-up here... are you still looking into whether to call this asynchronously?